securedrop/sdconfig.py

```diff
@@ -0,0 +1,174 @@
+            pass
+
+        try:
+            self.DATABASE_ENGINE = _config.DATABASE_ENGINE  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.DATABASE_FILE = _config.DATABASE_FILE  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.DEFAULT_LOCALE = _config.DEFAULT_LOCALE  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.GPG_KEY_DIR = _config.GPG_KEY_DIR  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.JOURNALIST_KEY = _config.JOURNALIST_KEY  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.JOURNALIST_TEMPLATES_DIR = _config.JOURNALIST_TEMPLATES_DIR  # type: ignore # noqa: E501
+        except AttributeError:
+            pass
+
+        try:
+            self.NOUNS = _config.NOUNS  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.SCRYPT_GPG_PEPPER = _config.SCRYPT_GPG_PEPPER  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.SCRYPT_ID_PEPPER = _config.SCRYPT_ID_PEPPER  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.SCRYPT_PARAMS = _config.SCRYPT_PARAMS  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.SECUREDROP_DATA_ROOT = _config.SECUREDROP_DATA_ROOT  # type: ignore # noqa: E501
+        except AttributeError:
+            pass
+
+        try:
+            self.SECUREDROP_ROOT = _config.SECUREDROP_ROOT  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.SESSION_EXPIRATION_MINUTES = \
+                _config.SESSION_EXPIRATION_MINUTES  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.SOURCE_TEMPLATES_DIR = \
+                _config.SOURCE_TEMPLATES_DIR  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.STORE_DIR = _config.STORE_DIR  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.SUPPORTED_LOCALES = \
+                _config.SUPPORTED_LOCALES  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.TEMP_DIR = _config.TEMP_DIR  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.WORD_LIST = _config.WORD_LIST  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.WORKER_PIDFILE = _config.WORKER_PIDFILE  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.TRANSLATION_DIRS = _config.TRANSLATION_DIRS  # type: ignore
+        except AttributeError:
+            pass
+
+        try:
+            self.env = _config.env  # type: ignore
```

Comment:
please do not include modules, only variables

Reply:
@dachary `env` is not a module, it is a variable of type `str`.

Reply:
correct, but it is only used internally and not anywhere else in the code so let's not needlessly expose it

Reply:
It's used for starting the app in the development environment and as a safeguard against listening on `0.0.0.0` in prod.

Reply:
ah, right it's
<pre>
    debug = getattr(config, 'env', 'prod') != 'prod'
</pre>
in journalist.py and source.py. It is worth a comment to point to these. I looked for `config.env` for usage and did not find anything, reason for my remark.

Reply:
^^ this is the exact reason I don't want any dynamic handling of attributes in the code base :+1:

Reply:
Pleaes add the following above `self.env` to avoid similar confusion when reading the code (because this discussion is unlikely to be linked):
<pre>
# although config.env is intended to be private, it needs to be exposed for journalist.py / source.py to know if it is running in production or not.  
</pre>